### ISSUE TITLE
Using keyof inside of generic results in SyntaxError failure

### ISSUE THREAD

**Issue by k2snowman69** [typescript-eslint/typescript-eslint#909]
**What code were you trying to parse?**

```ts
export const removeKeys = <T>(payload: T, keys: Array<keyof T>) => {
};
```

**What did you expect to happen?**
Everything to parse without error

**What actually happened?**
`SyntaxError: JSX element 'keyof' has no corresponding closing tag. (1:55)`

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `2.0.0` |
| `TypeScript`                           | `3.5.2` |
| `node`                                 | `10.16.0` |
| `npm`                                  | `6.11.2` |
